src/parser/shared/modules/spells/bfa/azeritetraits/GraceOfTheJusticar.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import Analyzer from 'parser/core/Analyzer';
+import { formatNumber } from 'common/format';
+import SPELLS from 'common/SPELLS/index';
+import TraitStatisticBox, { STATISTIC_ORDER } from 'interface/others/TraitStatisticBox';
+import ItemHealingDone from 'interface/others/ItemHealingDone';
+
+/**
+ * Grace of the Justicar
+ * Judging a foe heals up to 10 allies within 8 yards of that enemy for 899. (@iLvl 400)
+ * Example Log: https://www.warcraftlogs.com/reports/kMbVanmJwCg7WrAz#fight=last&type=summary&source=2
+ */
+class GraceOfTheJusticar extends Analyzer {
+  graceHealing = 0;
+  targetsHit = 0;
+  beaconTransfer = 0;
+  casts = 0;
+  justJudged = false;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.GRACE_OF_THE_JUSTICAR_TRAIT.id);
+  }
+
+  on_byPlayer_cast(event) {
```

Comment:
Have you considered using the manual event listening through `addEventListener`? This is sort of the preferred way (though the legacy method isn't deprecated yet).

Reply:
I have converted everything but the on_beacontransfer event.  I am attempting to add the event like this:
this.addEventListener(Events.beaconTransfer.by(SELECTED_PLAYER).source(SPELLS.GRACE_OF_THE_JUSTICAR), this.onBeaconTransfer);
But that is not correct.  I did a search and I don't see a beacontransfer listener being used.
Can you point me in the right direction?

Reply:
https://github.com/WoWAnalyzer/WoWAnalyzer/blob/106a68c71a4f1c0460b931348e270eed175060a8/src/parser/core/Events.js

Reply:
@HawkCorrigan thanks.  I found that also.  I guess I would have to add a filter for beacontransfer events?

Reply:
I tried to add a new Event like this:

`  get beacontransfer() {
    return new EventFilter('beacontransfer');
  },
`

but was unsuccessful.

Reply:
do you have a log with the trait at hand?

Reply:
https://www.warcraftlogs.com/reports/kMbVanmJwCg7WrAz#fight=last&type=summary&source=2

Reply:
Just tested it real quick

Technically adding the EventFilter works.
You propably went wrong when adding the eventListener
The beacontransfer still has Beacon as it's spell instead of Grace which means you can't filter it how you wrote it above.
Additionally it would need to be .spell instead of .source

![grafik](https://user-images.githubusercontent.com/2129161/54080014-d5998f00-42e7-11e9-835a-6407387e2997.png)

Reply:
so adding this:
```
  get beacontransfer() {
    return new EventFilter('beacontransfer');
  },
```
enables you to do 
```
this.addEventListener(Events.beacontransfer.by(SELECTED_PLAYER), this.onBeaconTransfer);
```

Reply:
@HawkCorrigan Thanks, that did the trick.